### DISCUSSION TITLE
Start app as floating

### DISCUSSION THREAD

**Discussion by patrikha** [leftwm/leftwm#665] [Q&A]
Hi,

Anyone know if its possible to start a new app as floating instead of tiled (from a script in polybar)? I'm in the process och creating rofi applets to the polybar and some utilities should start floated for best user experience.

**Comment by VuiMuich** [comment]
Hi,
there are basically two ways:
- you can create a [window rule](https://github.com/leftwm/leftwm/wiki/Config#window-rules) in `config.toml`
- if you are able to define the `window_type` of your app setting it to `Utility` should make it kind of 'getting ignored' by the WM

**Reply by patrikha** [reply]
Nice, window_rules are now implemented awesome!

But cant make it to work...
`
_NET_WM_DESKTOP(CARDINAL) = 3
WM_STATE(WM_STATE):
		window state: Normal
		icon window: <field not available>
_NET_WM_STATE(ATOM) =
WM_HINTS(WM_HINTS):
		Client accepts input or input focus: True
		Initial state is Normal State.
		bitmap id # to use for icon: 0x9e00024
		bitmap id # of mask for icon: 0x9e00025
		window id # of group leader: 0x9e00001
_MOTIF_WM_HINTS(_MOTIF_WM_HINTS) = 0x3, 0x13, 0x45, 0x0, 0x0
_NET_WM_ICON(CARDINAL) = 	Icon (128 x 128):
	(not shown)

WM_WINDOW_ROLE(STRING) = "Free42 Calculator"
_NET_WM_SYNC_REQUEST_COUNTER(CARDINAL) = 165675043
_NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_NORMAL
_NET_WM_USER_TIME_WINDOW(WINDOW): window id # 0x9e00022
WM_CLIENT_LEADER(WINDOW): window id # 0x9e00001
_NET_WM_PID(CARDINAL) = 534398
WM_LOCALE_NAME(STRING) = "en_US.UTF-8"
WM_CLIENT_MACHINE(STRING) = "mac-linux"
WM_NORMAL_HINTS(WM_SIZE_HINTS):
		program specified location: 0, 0
		program specified minimum size: 340 by 644
		program specified maximum size: 340 by 644
		window gravity: NorthWest
WM_PROTOCOLS(ATOM): protocols  WM_DELETE_WINDOW, WM_TAKE_FOCUS, _NET_WM_PING, _NET_WM_SYNC_REQUEST
WM_CLASS(STRING) = "free42bin", "Free42bin"
WM_ICON_NAME(STRING) = "Free42 Binary"
_NET_WM_ICON_NAME(UTF8_STRING) = "Free42 Binary"
WM_NAME(STRING) = "Free42 Binary"
_NET_WM_NAME(UTF8_STRING) = "Free42 Binary"
`

and testing with both:
`
  {window_class = "free42bin", spawn_floating = true},
  {window_title = "Free42 Binary", spawn_floating = true},
`

But still showing as tiled.

Using leftwm:
`
:: LeftWM version: 0.3.0
:: LeftWM git hash: 43a278a
`
